src/psyclone/psyir/transformations/parallel_loop_trans.py

```diff
@@ -45,15 +45,19 @@
 from psyclone.core import Signature
 from psyclone.domain.common.psylayer import PSyLoop
 from psyclone.psyir import nodes
-from psyclone.psyir.nodes import Loop, Reference, Call, Routine
+from psyclone.psyir.nodes import (
+        Call, Loop, Reference, Routine
+)
 from psyclone.psyir.symbols import AutomaticInterface
 from psyclone.psyir.tools import DependencyTools, DTCode
 from psyclone.psyir.transformations.loop_trans import LoopTrans
+from psyclone.psyir.transformations.async_trans_mixin import \
+    AsyncTransMixin
 from psyclone.psyir.transformations.transformation_error import \
     TransformationError
 
 
-class ParallelLoopTrans(LoopTrans, metaclass=abc.ABCMeta):
+class ParallelLoopTrans(LoopTrans, AsyncTransMixin, metaclass=abc.ABCMeta):
     '''
     Adds an abstract directive (it needs to be specified by sub-classing this
     transformation) to a loop indicating that it should be parallelised. It
```

Comment:
I think the new `nowait` option merits a sentence or two in this docstring - just to cover how it's intended to be used and where the resulting barrier will be placed.

Reply:
Added

Reply:
Thanks. Could you extend a bit to say where the barrier will be placed by default.

Reply:
Added